File: s/stack/Stack calculations.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AlgorithmsDataStructures
{
    public partial class Stack<T>
    {
        public static int StacksCalculations(string expression)
        {
            Stack<object> firstStack = new Stack<object>();
            Stack<int> secondStack = new Stack<int>();
            char[] chars = expression.ToCharArray();

            bool numberContinues = false;
            string number = "";
            for (int i = chars.Length - 1; i >= 0; --i)
            {
                if (numberContinues && char.IsDigit(chars[i])) number = chars[i] + number;
                else if (numberContinues)
                {
                    firstStack.Push(int.Parse(number));
                    number = "";
                    numberContinues = false;
                }
                else if (char.IsDigit(chars[i]))
                {
                    numberContinues = true;
                    number += chars[i];
                }
                else if (chars[i] != ' ') firstStack.Push(chars[i]);
            }

            firstStack.Push(int.Parse(number));

            while (firstStack.Size() > 0)
            {
                object currentObject = firstStack.Pop();
                char.TryParse(currentObject.ToString(), out char currentElement);

                if (currentObject is int) secondStack.Push((int)currentObject);
                else if (currentElement == '=') return secondStack.Pop();
                else
                {
                    int secondNumber = secondStack.Pop();
                    int firstNumber = secondStack.Pop();
                    switch (currentElement)
                    {
                        case '+':
                            secondStack.Push(firstNumber + secondNumber);
                            break;
                        case '*':
                            secondStack.Push(firstNumber * secondNumber);
                            break;
                        case '-':
                            secondStack.Push(firstNumber - secondNumber);
                            break;
                        case '/':
                            secondStack.Push(firstNumber / secondNumber);
                            break;
                        default:
                            throw new ArgumentException("Wrong input expression");
                    }
                }
            }
            return 0;
        }
    }
}

File: 04.stack/stack/python/stack_calculations.py
from stack import Stack


def stacks_calculations(expression):
    first_stack = Stack()
    second_stack = Stack()
    chars = list(expression)

    number_continues = False
    number = ""

    for i in range(len(chars) - 1, -1, -1):
        if number_continues and chars[i].isdigit():
            number = chars[i] + number
        elif number_continues:
            first_stack.push(int(number))
            number = ""
            number_continues = False
        elif chars[i].isdigit():
            number_continues = True
            number += chars[i]
        elif chars[i] != ' ':
            first_stack.push(chars[i])

    if number:
        first_stack.push(int(number))

    while first_stack:
        current_object = first_stack.pop()

        if isinstance(current_object, int):
            second_stack.push(current_object)
        elif current_object == '=':
            return second_stack.pop()
        else:
            second_number = second_stack.pop()
            first_number = second_stack.pop()
            if current_object == '+':
                second_stack.push(first_number + second_number)
            elif current_object == '*':
                second_stack.push(first_number * second_number)
            elif current_object == '-':
                second_stack.push(first_number - second_number)
            elif current_object == '/':
                second_stack.push(first_number / second_number)
            else:
                raise ValueError("Wrong input expression")

    return 0


print(stacks_calculations(input()))
